packages/vuepress-theme-odyssey/components/DocsSidebar.vue

```diff
@@ -0,0 +1,150 @@
+      :return-focus-on-deactivate="true"
+      :click-outside-deactivates="true"
+      :initial-focus="() => $refs.closeButton"
+    >
+      <div class="docs-sidebar">
+        <div class="docs-sidebar--content">
+          <button
+            ref="closeButton"
+            class="docs-sidebar--action ods-button is-ods-button-clear"
+            aria-label="Close Navigation"
+            @click="setSidebarState(false)"
+          >
+            <OdsIcon icon="close" />
+          </button>
+
+          <div class="docs-sidebar--header">
+            <DocsLink class="docs-site-title" href="/">{{ title }}</DocsLink>
+            <fieldset v-if="showSearch" class="ods-fieldset">
+              <div class="ods-fieldset-flex">
+                <input
+                  id="search"
+                  class="ods-text-input"
+                  type="search"
+                  name="search"
+                  autocomplete="search"
+                  spellcheck="false"
+                  placeholder="Search"
+                  required
+                />
+                <label class="ods-label" for="search">Search</label>
+              </div>
+            </fieldset>
+          </div>
+          <div
+            :class="{
+              'docs-sidebar--main': true,
+              'is-overflowing': isOverflowing
+            }"
+          >
+            <div ref="mainContent" class="docs-sidebar--main-content">
+              <DocsNav variant="primary" :nav="nav.primary" />
+            </div>
+          </div>
+          <div class="docs-sidebar--footer">
+            <DocsNav variant="secondary" :nav="nav.secondary" />
+          </div>
+        </div>
+      </div>
+    </FocusTrap>
+  </Fragment>
+</template>
+
+<script>
+import { Fragment } from "vue-fragment";
+import { FocusTrap } from "focus-trap-vue";
+
+export default {
+  name: "DocsSidebar",
+  components: {
+    Fragment,
+    FocusTrap,
+    DocsLink: () => import("./DocsLink.vue"),
+    DocsNav: () => import("../components/DocsNav.vue"),
+    DocsTopbar: () => import("../components/DocsTopbar.vue")
+  },
+  props: {
+    showSearch: {
+      type: Boolean,
+      default: false
+    },
+    title: {
+      type: String,
+      default: "default"
+    },
+    nav: {
+      type: Object,
+      default: () => ({})
+    }
+  },
+  data: () => ({
+    isOverflowing: null,
+    isExpanded: false
+  }),
+  watch: {
+    isExpanded: function(isExpanded) {
+      const el = document.documentElement;
+      const className = "is-sidebar-expanded";
+
+      if (isExpanded) {
+        el.classList.add(className);
+
+        // TODO: Figure out a better solution here, this
+        // is required because focus isnt always applied to
+        // the close button when the element is expanded due to
+        // `visibility:hidden;` being applied previously.
+        setTimeout(() => this.$refs.closeButton.focus(), 100);
+      } else {
+        el.classList.remove(className);
```

Comment:
In React, there's a small tool called `classnames` (https://github.com/JedWatson/classnames) that will add or remove classes based on a state variable. It was very useful and handy in enduser.

Reply:
I love `classNames`! We use `classNames` like syntax in various components, this one was a little different as it changes the class within on the HTML element, so it wouldn't apply in this specific instance.